src/data/providers/balancer-api/modules/nested-pool-state/index.ts

```diff
@@ -258,14 +248,25 @@ export function mapPoolToNestedPoolStateV2(pool: PoolGetPool): NestedPoolState {
         });
     });
 
-    const mainTokens = pool.allTokens
-        .filter((t) => t.isMainToken)
-        .map((t) => {
-            return {
-                address: t.address,
-                decimals: t.decimals,
-            };
-        });
+    // mainTokens are pool tokens filtering out nested pools and phantomBPTs
+    const mainTokens = pool.poolTokens.flatMap((t) => {
+        if (t.nestedPool) {
```

Comment:
Think you need to filter the phantomBPT first, otherwise you get the mainTokens of the parent pool twice. 

Also think about duplicated tokens. A usdc/bb-a-usd pool would have 2x usdc, do you need it twice or just once?

Reply:
You're right about filtering out phantomBPT from the parent pool - I'll fix that.
But we don't have to worry about duplicate tokens as that's not supported by the SDK - it will fail in the validator. There shouldn't exist such a pool composition.

Reply:
I changed the logic a bit to make it simpler and easier to read/understand

Reply:
Looks good!